eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -106,7 +106,47 @@ if ($LatestsSorted[0] -ne $Version) {
 
 if ($ReplaceLatestEntryTitle)
 {
-    $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus -Content $ChangeLogEntries[$LatestVersion].ReleaseContent
+    # Remove empty sections from content
+    $sanitizedContent = @()
+    $sectionContent = @()
+    $sectionContentCount = 0
+    $latesVersionContent = $ChangeLogEntries[$LatestVersion].ReleaseContent
+
+    foreach ($line in $latesVersionContent)
+    {
+        if ($line.StartsWith("### ") -or $sectionContentCount -gt 0)
```

Comment:
We should avoid parsing the sections again and instead use the Sections dictionary we created already (https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/ChangeLog-Operations.ps1#L59). We should loop through those sections and if any of those are empty we should find that line in the content and strip it and any empty lines under it.

Reply:
We should also move this logic into the ChangeLog-Operations.ps1 so we can be sure it stays in sync with the other parsing of sections we do there.

Reply:
Am not really parsing it again, instead it is blindly removing empty sections including those that are not amond the recomended sections. I think this is a more straight forward approach than using the parsed sections. The parsed sections are only useful in the context of vaildation, when writing the changelog we make use of the full releasecontent which is what I cleaned up here to remove the empty sections